### PR TITLE
Ship a basic 404 page

### LINKED ISSUE #72: Explore shipping a 404.rst
🐞 **Problem**
When a page or link is broken, it would be interesting to show a custom 404 page. For this purpose, the [notfound.extension](https://pypi.org/project/sphinx-notfound-page/) is used.

However, previous extensions requires to pass the path to a 404.rst file:

```python
notfound_template = "404.rst"
```

This situation forces projects need to store a custom `404.rst` file, see [PyMAPDL custom 404.rst file](https://raw.githubusercontent.com/pyansys/pymapdl/main/doc/source/404.rst).

💡 **Solution**
It would be a neat feature being able to import previous file from the `ansys-sphinx-theme`:

```python
from ansys_sphinx_theme import ansys_404_template

...

notfound_template = ansys_404_template
```

### PR DESCRIPTION
**Pull request by jorgepiloto**
Resolves #72 by including a basic 404 page.

### CODE REVIEW

File: src/ansys_sphinx_theme/static/404.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+Oops!
+=====
+This is unexpected. The page you are requesting does not exist.
+
+If this page should exist, please contact `pyansys.support@ansys.com <mailto:pyansys.support@ansys.com>`_.
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/static/404.rst:7]: I am not sure if there is a support for Ansys related projects. All public projects within Ansys organization contain references to personal email or PyAnsys support.

Comment (RobPasMue) [reply src/ansys_sphinx_theme/static/404.rst:7]: Honestly, this is out of the PyAnsys org. Anybody could potentially use this Sphinx theme for their own local projects. So, I'd recommend to not provide any support email. In fact, I'd suggest trying to link somehow to the Issues page of the GitHub repo associated to the build. For example, if PyMAPDL docs are using the Sphinx theme, this should link to the Issues page of PyMAPDL. I don't know if it'd be possible, but that would be the ideal scenario.

**Review by RobPasMue — APPROVED** [review]
Apart from my previous comment saying that we should target the issues page of each specific repo (if possible), nothing else. LGTM!

File: src/ansys_sphinx_theme/static/404.rst (hunk 1 above)
Comment (jorgepiloto) [reply src/ansys_sphinx_theme/static/404.rst:7]: From private discussion, let us investigate the `setup(app)` function to extract the `github_url` variable from the `html_theme_options`.

File: doc/source/usage.rst [hunk 2]
```diff
@@ -186,3 +186,60 @@ import and add them in the ``latex_additional_files``:
...
+
+Installing and enabling ``sphinx-notfound-page``
+------------------------------------------------
+
+.. code-block:: text
+
+    python -m pip install sphinx-notfound-page
+
+Consider adding this extension to your ``requirements_doc.txt`` or similar too.
+After that, enable the extension in ``conf.py`` file:
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply doc/source/usage.rst:206]: I reads "too", but I see it is confusing. Let me remove those last words.

[the remark this answers is not in the document]
Comment (jorgepiloto) [reply doc/source/usage.rst]: Fixed in https://github.com/ansys/ansys-sphinx-theme/pull/83/commits/7f371b70c561a20e578d720f51b8085a9cdf3255 👍🏽

File: src/ansys_sphinx_theme/static/404.rst (hunk 1 above)
Comment (jorgepiloto) [reply src/ansys_sphinx_theme/static/404.rst:7]: I forgot about this part... It turns out that we can use Jinja syntax `{{ theme_github_url }}` 😍

Comment (RobPasMue) [reply src/ansys_sphinx_theme/static/404.rst:7]: That's just great! Thanks @jorgepiloto

File: src/ansys_sphinx_theme/static/404.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+Oops!
+=====
+This is unexpected. The page you are requesting does not exist.
+
+If this page should exist, please contact `{{ theme_contact_mail }} <mailto:{{ theme_contact_mail }}`_.
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/static/404.rst:7]: Considering that some projects may not be linked to GitHub (see #38), we agreed on adding a new theme variable named `contact_mail` which is used in the `404.rst` file.

Comment (RobPasMue) [reply src/ansys_sphinx_theme/static/404.rst:7]: #OpeningPRsEverywhere

**Comment by RobPasMue** [comment]
Quick question @jorgepiloto - what's missing in this PR? Is it ready to merge?

**Comment by jorgepiloto** [comment]
> Quick question @jorgepiloto - what's missing in this PR? Is it ready to merge?

Nothing else to be added from my side. This is ready to be merged 👍🏽

**Comment by RobPasMue** [comment]
Merging then!

**Pull request merged**